src/ol/style/IconImage.js

```diff
@@ -75,27 +75,23 @@ class IconImage extends EventTarget {
 
     /**
      * @private
-     * @type {boolean|undefined}
      */
     this.tainted_;
   }
 
   /**
    * @private
-   * @param {CanvasRenderingContext2D=} context A context with the image already drawn into.
    * @return {boolean} The image canvas is tainted.
    */
-  isTainted_(context) {
+  isTainted_() {
     if (this.tainted_ === undefined && this.imageState_ === ImageState.LOADED) {
-      if (!context) {
-        context = createCanvasContext2D(1, 1);
-        context.drawImage(this.image_, 0, 0);
-      }
+      const context = createCanvasContext2D(1, 1);
+      context.drawImage(this.image_, 0, 0);
       try {
         context.getImageData(0, 0, 1, 1);
-        this.tainted_ = false;
-      } catch (e) {
         this.tainted_ = true;
+      } catch (e) {
+        this.tainted_ = false;
```

Comment:
I don't think this change is intentional, is it?

Reply:
Good question, yes it is but for me, its relay a strange change, If I don't do that it's not working on my Firefox, did I change the logic somewhere?

For me, this function should return true if the Canvas is working and some other..., and with the old code it's the inverse. I'm probably wrong somewhere but where?

Reply:
@sbrunner The `catch` block is meant to handle the the case where the canvas cannot be accessed because it is tainted. Better debug the try-block to see if you're running into the catch lock for a different reason. But the content of the catch-block should not be changed.

Reply:
Effectively, I get it back and it's working :-) Thanks :-)